azure-pipelines.yml

```diff
@@ -42,8 +42,10 @@ stages:
     steps:
     - script: make arm_sdk_install
       displayName: 'Install the build toolchain'
-    - script: sudo apt-get install -y libblocksruntime-dev
+    - script: sudo apt-get install -y libblocksruntime-dev astyle
       displayName: 'Install extra packages'
+    - bash: for f in `git diff --name-only`; do src/utils/check_cstyle.sh "${f}"; done
```

Comment:
Not sure this will work - since the repository is checked out into the branch at this point, the `git diff --name-only` will return an empty list.
A better way will be `git diff --name-only master`, but this has shortcomings as well:
- spurious files if there were commits to `master` after this branch was branched;
- a more or less arbitrary list of files if the pull request is against a different branch than `master`;
- obviously this will not work for CI builds against `master`, but I think this case can be ignored (or better: running of style checks can be suppressed in this case).

Reply:
How about: git diff --name-only "$SYSTEM_PULLREQUEST_SOURCEBRANCH", according to
[Migrate from Travis to Azure Pipelines](https://docs.microsoft.com/en-us/azure/devops/pipelines/migrate/from-travis?view=azure-devops), that is "The name of the branch where the pull request originated."

Regarding separate job, that is very good. I will put this in a separate job, e.g.:

```
 - job: 'Style check'
    pool:
      vmImage: 'ubuntu-16.04'
    steps:
    - script: sudo apt-get install -y astyle
      displayName: 'Install extra packages'
    - bash: for f in `git diff --name-only "$SYSTEM_PULLREQUEST_SOURCEBRANCH"`; do src/utils/check_cstyle.sh "${f}"; done
      displayName: 'C style check'
```

Reply:
@tstibor: Sorry, only picked this up again now.
Was there a problem with the above approach? The current version of `git diff HEAD HEAD~ --name-only` has got the issue that it will miss some files if the pull request contains more than one commit.